src/client/Client.cc

```diff
@@ -5518,10 +5518,10 @@ void Client::handle_cap_export(MetaSession *session, Inode *in, const MConstRef<
         if (it != in->caps.end()) {
 	  Cap &tcap = it->second;
 	  if (tcap.cap_id == m->peer.cap_id &&
-	      ceph_seq_cmp(tcap.seq, m->peer.seq) < 0) {
+	      ceph_seq_cmp(tcap.seq, m->peer.issue_seq) < 0) {
 	    tcap.cap_id = m->peer.cap_id;
-	    tcap.seq = m->peer.seq - 1;
-	    tcap.issue_seq = tcap.seq;
+	    tcap.seq = m->peer.issue_seq - 1;
+	    tcap.issue_seq = tcap.issue_seq;
```

Comment:
@batrick I was taking a look at "ceph: correct ceph_mds_cap_peer field name" patch for the kernel client and there is something fishy here:
- previously both clients effectively assigned `m->peer.seq - 1` to `tcap.seq` and `tcap.issue_seq`
- `m->peer.seq` got renamed to `m->peer.issue_seq`
- kernel client continues to assign that value to both fields
```
			tcap->seq = t_issue_seq - 1;
			tcap->issue_seq = t_issue_seq - 1;
```
- userspace client assigns that value only to `tcap.seq`, for `tcap.issue_seq` there appears to be a bogus self-assignment

Reply:
> @batrick I was taking a look at "ceph: correct ceph_mds_cap_peer field name" patch for the kernel client and there is something fishy here:
> 
>     * previously both clients effectively assigned `m->peer.seq - 1` to `tcap.seq` and `tcap.issue_seq`

Correct.

>     * `m->peer.seq` got renamed to `m->peer.issue_seq`

Correct.

>     * kernel client continues to assign that value to both fields
> 
> 
> ```
> 			tcap->seq = t_issue_seq - 1;
> 			tcap->issue_seq = t_issue_seq - 1;
> ```
> 
>     * userspace client assigns that value only to `tcap.seq`, for `tcap.issue_seq` there appears to be a bogus self-assignment

Oh, very nice catch. My find/replace clearly messed that up.